File: testdata/jennies/builders/builder_delegation_in_disjunction/PHPBuilder/src/BuilderDelegationInDisjunction/DashboardLinkBuilder.php
<?php

namespace Grafana\Foundation\BuilderDelegationInDisjunction;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink>
 */
class DashboardLinkBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BuilderDelegationInDisjunction\DashboardLink
     */
    public function build()
    {
        return $this->internal;
    }

    public function title(string $title): static
    {
        $this->internal->title = $title;
    
        return $this;
    }

    public function url(string $url): static
    {
        $this->internal->url = $url;
    
        return $this;
    }

}

File: testdata/jennies/builders/builder_delegation_in_disjunction/JavaBuilder/builder_delegation_in_disjunction/DashboardLinkBuilder.java
package builder_delegation_in_disjunction;


public class DashboardLinkBuilder implements cog.Builder<DashboardLink> {
    protected final DashboardLink internal;
    
    public DashboardLinkBuilder() {
        this.internal = new DashboardLink();
    }
    public DashboardLinkBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    
    public DashboardLinkBuilder url(String url) {
        this.internal.url = url;
        return this;
    }
    public DashboardLink build() {
        return this.internal;
    }
}
